### PR TITLE
[Script.extendedinfo@matrix] 6.0.1

### PR DESCRIPTION
**Pull request by scott967**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

This is a reboot of script.extendedinfo currently in repo-scripts.  I have been unable to contact original author Phil65; however, I do not believe he is active at this time.

The main purpose of this release is to update it for use in Kodi Matrix/Nexus including python3 compatibility.  However, due to various changes required due to changes in the online data sources since this addon was last updated I have had to make numerous updates including but not limited to:

- update dependency on script.module.kodi65 to script.module.kutils
- add settings for user-supplied api keys
- adjust some api calls
- remove support for bandsintown (addon not eligible for api key per site TOU)
- refactor code to PEP8
- add doctrings (WIP)
- change ownership to scott967

To facilitate review I have provided 2 commits:
1.  all changes up to last version in phil65 source (5.6.0) note: slight discrepancy with repo-scripts
2.  my changes from 5.6.0

If needed these can be squashed.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Auto detect text files and perform LF normalization
```
Comment (basrieter) [inline .gitattributes:1]: Remove this file from your PR

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# vs code ignores
```
Comment (basrieter) [inline .gitignore:1]: Remove this file from your PR

File: .gitattributes (hunk 1 above)
Comment (scott967) [reply .gitattributes]: Should be fixed now

File: .gitignore (hunk 2 above)
Comment (scott967) [reply .gitignore]: Should be fixed now

**Comment by basrieter** [comment]
@phil65 are you OK with this new version?

**Comment by scott967** [comment]
Any chance getting a review on this?

**Comment by scott967** [comment]
I made some minor changes, adding some docstrings and type annotations and in a couple places changed string formatting to f-strings to improve readability.  Then squashed all into a single commit.

**Comment by basrieter** [comment]
> I made some minor changes, adding some docstrings and type annotations and in a couple places changed string formatting to f-strings to improve readability. Then squashed all into a single commit.

I would still like an OK from the original author: @phil65

**Comment by thezoggy** [comment]
> > I made some minor changes, adding some docstrings and type annotations and in a couple places changed string formatting to f-strings to improve readability. Then squashed all into a single commit.
> 
> I would still like an OK from the original author: @phil65

you've asked for an OK from the original author for 2+ months now.. what is the max limbo time to wait?

**Comment by scott967** [comment]
Please submit support requests in Kodi skin development forum
If kutils or any other addon fails to install from mirrors, retry later.
The exception in kutils is due to an update ver 2.1.1 of script.module.autocompletion which is not backward compatible and breaks addons which use that script.  I have submitted a PR to the author that should fix it.  In the meantime a work around is to install plugin.program.autocompletion from the Kodi repo "programs" addon node.

**Pull request merged**